### PR TITLE
Improve technical documentation for dip test and event detection

### PR DESCRIPTION
**Pull request by greaa-aws**
### Description
This PR improves documentation for the dip test and event detection components of the metrics correlation feature.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch_py_ml/ml_models/metrics_correlation/dip.py [hunk 1]
```diff
@@ -12,29 +12,26 @@
...
     """
-    Fast computation of the dip test statistic. See: https://www.jstor.org/stable/pdf/2241144.pdf (paper)
-    and https://www.jstor.org/stable/pdf/2347485.pdf (pseudocode)
-    Debugged and validated against the R package of Prof. Martin Maechler, ETH Zurich:
-    https://rdrr.io/cran/diptest/
+    Computes the Dip test statistic from a sequence of real-valued
+    observations.
 
-    :param x:
+    :param x: 1-D sequence to be evaluated for unimodality.
```
Comment (dhrubo-os) [inline opensearch_py_ml/ml_models/metrics_correlation/dip.py:18]: can we rename the variable with more meaning name?

File: opensearch_py_ml/ml_models/metrics_correlation/dip.py [hunk 2]
```diff
@@ -12,29 +12,26 @@
...
     :type x: torch.Tensor
-    :return:
+    :return: The dip statistic $d$.
```
Comment (dhrubo-os) [inline opensearch_py_ml/ml_models/metrics_correlation/dip.py:20]: what is `$d$`? Is this unicode or ascii?

File: opensearch_py_ml/ml_models/metrics_correlation/dip.py [hunk 3]
```diff
@@ -12,29 +12,26 @@
...
-            0.0  # constant is technically unimodal, but we want to reject these events
-        )
+        return 0.0
 
     n = len(x)
     low = 0
     high = n - 1
 
-    # establish indices mn over which combination is req's for GCM fit
+    # establish indices mn over which combination is req'd for GCM fit
```
Comment (dhrubo-os) [inline opensearch_py_ml/ml_models/metrics_correlation/dip.py:34]: Can we please elaborate GCM? Same goes for LCM

File: opensearch_py_ml/ml_models/metrics_correlation/dip.py [hunk 4]
```diff
@@ -381,19 +394,22 @@ def diptest(x: torch.Tensor) -> Tuple[float, float]:
...
 def interp(x: torch.Tensor, xp: torch.Tensor, fp: torch.Tensor) -> float:
     """
-    Intercept
+    Simple interpolation scheme to estimate the value at `x` of a function
+    with values `fp` at points `xp`. If `x` falls within the range of `xp`,
+    use linear interpolation. Otherwise use constant interpolation from the
+    nearest observed point.
 
-    :param x:
+    :param x: Point at which to approximate the function.
```
Comment (dhrubo-os) [inline opensearch_py_ml/ml_models/metrics_correlation/dip.py:402]: what do you think about renaming the variables?

File: opensearch_py_ml/ml_models/metrics_correlation/dip.py (hunk 1 above)
Comment (greaa-aws) [reply opensearch_py_ml/ml_models/metrics_correlation/dip.py:18]: Sure. I'll make a change here and for the comment on `interp()` below.

File: opensearch_py_ml/ml_models/metrics_correlation/dip.py (hunk 2 above)
Comment (greaa-aws) [reply opensearch_py_ml/ml_models/metrics_correlation/dip.py:20]: $ denotes math notation. Supported in many markdown formats, e.g. in github comments it renders as $d$ (but also helps you write more complex expressions like $\exp^{\frac{-x^2}{\sigma}}$). 

Not sure if it's supported for our documentation. I can remove it if not.

File: opensearch_py_ml/ml_models/metrics_correlation/dip.py (hunk 3 above)
Comment (greaa-aws) [reply opensearch_py_ml/ml_models/metrics_correlation/dip.py:34]: Sure. I can add an additional level of detail here.

File: opensearch_py_ml/ml_models/metrics_correlation/dip.py [hunk 5]
```diff
@@ -379,29 +396,32 @@ def diptest(x: torch.Tensor) -> Tuple[float, float]:
...
-    :return:
+    Simple interpolation scheme to estimate the value at `x_new` of a function
+    with values `f_vals` at points `x_vals`. If `x_new` falls within the range of
+    `x_vals`, use linear interpolation. Otherwise use constant interpolation from the
+    nearest observed point.
+
+    :param x_new: Point at which to approximate the function.
+    :type x_new: torch.Tensor
+    :param x_vals: Observed input values.
+    :type xx_valsp: torch.Tensor
```
Comment (dhrubo-os) [inline opensearch_py_ml/ml_models/metrics_correlation/dip.py:409]: [nit] param `x_vals` is not matching with type `xx_valsp`

**Review by dhrubo-os — APPROVED** [review]

**Pull request merged**